src/Library/demos/Column View/main.js

```diff
@@ -0,0 +1,123 @@
+  factory.connect("setup", (factory, list_item) => {
+    const label = new Gtk.Label({
+      margin_start: 12,
+      margin_end: 12,
+    });
+    list_item.set_child(label);
+  });
+  factory.connect("bind", (factory, list_item) => {
+    const label_widget = list_item.get_child();
+    const model_item = list_item.get_item();
+    label_widget.label = model_item.get_display_name();
+  });
+}
+
+function createSecCol() {
+  const factory = col2.factory;
+  factory.connect("setup", (factory, list_item) => {
+    const label = new Gtk.Label({
+      margin_start: 12,
+      margin_end: 12,
+    });
+    list_item.set_child(label);
+  });
+  factory.connect("bind", (factory, list_item) => {
+    const label_widget = list_item.get_child();
+    const model_item = list_item.get_item();
+    label_widget.label = model_item.get_size().toString();
+  });
+}
+
+function createThirdCol() {
+  const factory = col3.factory;
+  factory.connect("setup", (factory, list_item) => {
+    const label = new Gtk.Label({
+      margin_start: 12,
+      margin_end: 12,
+    });
+    list_item.set_child(label);
+  });
+  factory.connect("bind", (factory, list_item) => {
+    const info = list_item.item;
+    const mtime = info.get_modification_date_time();
+    const date = new Date(mtime.to_unix() * 1000);
+    const now = new Date();
+
+    const options = {};
+    if (date.getMonth() < now.getMonth() || date.getDay() < now.getDay())
+      options.dateStyle = "long";
+    else options.timeStyle = "short";
+
+    const label = list_item.child;
+    label.label = Intl.DateTimeFormat(undefined, options).format(date);
+  });
+}
+
+//Model
+const dir = Gio.File.new_for_path(pkg.pkgdatadir).resolve_relative_path(
+  "Library/demos",
+);
+
+const model = new Gtk.SingleSelection({
+  model: new Gtk.SortListModel({
+    model: new Gtk.DirectoryList({
+      file: dir,
+      attributes: "standard::*,time::modified",
+    }),
+    sorter: column_view.sorter,
+  }),
+});
+
+col1.sorter = new Gtk.StringSorter({
+  expression: new Gtk.ClosureExpression(
+    GObject.TYPE_STRING,
+    (fileInfo) => fileInfo.get_display_name(),
+    null,
+  ),
+});
+
+col2.sorter = new Gtk.NumericSorter({
+  expression: new Gtk.ClosureExpression(
+    GObject.TYPE_INT,
+    (fileInfo) => fileInfo.get_size(),
+    null,
+  ),
+});
+
+col3.sorter = new Gtk.NumericSorter({
+  expression: new Gtk.ClosureExpression(
+    GObject.TYPE_INT64,
+    (fileInfo) => fileInfo.get_modification_date_time().to_unix(),
+    null,
+  ),
+});
+
+column_view.model = model;
+createFirstCol();
+createSecCol();
+createThirdCol();
+
+//View
+/*model.model.connect("items-changed", (list, position, removed, added) => {
+  console.log(
+    `position: ${position}, Item removed? ${Boolean(
+      removed,
+    )}, Item added? ${Boolean(added)}`,
+  );
+});
+
+model.connect("selection-changed", () => {
+  const selected_item = model.get_selected();
+  console.log(`Model item selected from view: ${model.model}`);
+});*/
```

Comment:
```suggestion
```

We won't bother with demonstrating the selection model itself, we did that in a previous demo.